### PR TITLE
Add: error handling, prevent notus from stopping

### PR DESCRIPTION
**Pull request by Kraemii**
**What**:
This PR improves the error handling to prevent it from crashing during scans.
SC-366

**Why**:
To improve the stability of the notus-scanner

**How**:
Using a MQTT tool to send scan start scan messages to the notus-scanner. I used MQTT explorer, but Mosquitto also comes with some command line tools.

Acceptance Criteria:
- It should be possible to issue a start scan message with an unknown or missing OS release without crashing notus-scanner
- It should be possible to issue a start scan message with missing package list without crashing notus-scanner
- Corrupt package names in the package list of the start scan message must be ignored
- Missing required information in all messages must be logged and the messages must be ignored
- Failures during the MQTT communication must not crash the daemon
- Failures while loading and parsing the JSON advisory file must be logged and must not result in a crash of the daemon

Message preset:
```json
{
    "message_id": "6072f686-5283-4b04-a87f-2b7c047961bb",
    "group_id": "9b9d2042-14f4-4630-8e10-90736075a13c",
    "message_type": "scan.start",
    "created": 1641404959,
    "scan_id": "97079ee9-8917-49da-aa4f-4ef95f757ac1",
    "host_ip": "127.0.0.1",
    "host_name": null,
    "os_release": "Debian 10",
    "package_list": [
        "firefox-esr-78.15.0esr-1~deb10u1\r"
    ]
}
```

Please try to break notus :)

**Checklist**:

- [x] Tests
- [ ] [CHANGELOG](https://github.com/greenbone/notus-scanner/blob/master/CHANGELOG.md) Entry
- [ ] Documentation

### CODE REVIEW

File: tests/messages/test_message.py [hunk 1]
```diff
@@ -137,3 +137,76 @@ def test_load(self):
...
+
+    def test_load_message_id_unvalid(self):
+        payload = (
+            '{"message_id": "foo", '
+            '"message_type": "scan.start", '
+            '"group_id": "866350e8-1492-497e-b12b-c079287d51dd", '
+            '"created": 1628512774.0}'
+        )
+        Message.message_type = MessageType.SCAN_START
+        self.assertRaises(ValueError, Message.load, payload)
```
Comment (bjoernricks) [inline tests/messages/test_message.py:149]: IMHO using the with statement for assertRaises is much more readable
```suggestion
        with self.assertRaises(ValueError):
            Message.load(payload)
```

**Review by bjoernricks — CHANGES_REQUESTED** [review]

File: notus/scanner/messages/message.py [hunk 2]
```diff
@@ -55,12 +55,26 @@ def _parse(cls, data: Dict[str, Union[int, str]]) -> Dict[str, Any]:
...
-            "message_id": UUID(data.get("message_id")),
-            "group_id": UUID(data.get("group_id")),
-            "created": datetime.fromtimestamp(
+
+        try:
+            message_id = UUID(data.get("message_id"))
+        except (TypeError, ValueError) as e:
+            raise e.__class__(f"error while parsing 'message_id', {e}")
+        try:
+            group_id = UUID(data.get("group_id"))
```
Comment (nichtsfrei) [inline notus/scanner/messages/message.py:64]: group_id is not a UUID

**Review by nichtsfrei — CHANGES_REQUESTED** [review]

File: notus/scanner/messages/message.py (hunk 2 above)
Comment (Kraemii) [reply notus/scanner/messages/message.py:64]: Yes it is. At least we always took one and it is also documented as one

File: notus/scanner/messages/message.py [hunk 3]
```diff
@@ -55,12 +55,26 @@ def _parse(cls, data: Dict[str, Union[int, str]]) -> Dict[str, Any]:
...
                 float(data.get("created")), timezone.utc
-            ),
+            )
+        except (TypeError, ValueError) as e:
+            raise e.__class__(f"error while parsing 'created', {e}")
+
+        return {
+            "message_id": message_id,
+            "group_id": group_id,
+            "created": created,
```
[the remark this answers is not in the document]
Comment (Kraemii) [reply notus/scanner/messages/message.py]: Replaced all Errors occur while parsing by `MessageParsingError`

**Review by bjoernricks — APPROVED** [review]

File: notus/scanner/messages/message.py (hunk 2 above)
Comment (nichtsfrei) [reply notus/scanner/messages/message.py:73]: It is not: https://github.com/greenbone/eulabeia/blob/main/docs/messaging.md but as discussed in the daily it is fine when it is a second pr.

**Review by nichtsfrei — APPROVED** [review]

**Pull request merged**